src/rosdep2/installers.py

```diff
@@ -368,7 +368,10 @@ def get_packages_to_install(self, resolved, reinstall=False):
         if not resolved:
             return []
         else:
-            return [x for x in resolved if x not in self.detect_fn(resolved)]
+            detected = set(self.detect_fn(resolved))
+            seen = set()
+            seen_add = seen.add
```

Comment:
Why do you need to do this? It's the same number of characters.

Reply:
For execution speed. [Reference](https://stackoverflow.com/a/480227).

Perhaps it is overkill.

Reply:
It's fine, I just wasn't aware of why. You might want a comment, otherwise someone might just remove it as redundant.